crypto/x509/v3_purp.c

```diff
@@ -48,9 +48,9 @@ static X509_PURPOSE xstandard[] = {
     {X509_PURPOSE_NS_SSL_SERVER, X509_TRUST_SSL_SERVER, 0,
      check_purpose_ns_ssl_server, "Netscape SSL server", "nssslserver", NULL},
     {X509_PURPOSE_SMIME_SIGN, X509_TRUST_EMAIL, 0, check_purpose_smime_sign,
-     "S/MIME signing", "smimesign", NULL},
+     "S/MIME signing", "smime_sign", NULL},
     {X509_PURPOSE_SMIME_ENCRYPT, X509_TRUST_EMAIL, 0,
-     check_purpose_smime_encrypt, "S/MIME encryption", "smimeencrypt", NULL},
+     check_purpose_smime_encrypt, "S/MIME encryption", "smime_encrypt", NULL},
```

Comment:
I do not think we can change these strings in a stable branch, as they are part of the public API.
(I am not convinced we want to change them even in a major release, as the benefit seems small compared to the churn in consumers of them.)
I do see that the `smime_sign` is listed in the documentation, but in this case I think the actual behavior of the code takes precedence.